# HG changeset patch
# User Markus Bergholz <markuman+octave@gmail.com>
# Date 1412877669 -7200
#      Thu Oct 09 20:01:09 2014 +0200
# Node ID aea099cd1ee751206b932c4320686ba3aec9641f
# Parent  52ae096f038b5002dc59722667b5cb1b0fe9532c
Allow stem to plot logical values (bug #43391).

* __stem__.m: Convert logical input to double before calling addproperty.

diff --git a/scripts/plot/draw/private/__stem__.m b/scripts/plot/draw/private/__stem__.m
--- a/scripts/plot/draw/private/__stem__.m
+++ b/scripts/plot/draw/private/__stem__.m
@@ -123,17 +123,23 @@ function h = __stem__ (have_z, varargin)
       addlistener (hg, "color", @update_props);
       addlistener (hg, "linestyle", @update_props);
       addlistener (hg, "linewidth", @update_props);
       addlistener (hg, "marker", @update_props);
       addlistener (hg, "markeredgecolor", @update_props);
       addlistener (hg, "markerfacecolor", @update_props);
       addlistener (hg, "markersize", @update_props);
 
+      if (islogical (x))
+        x = double (x);
+      endif
       addproperty ("xdata", hg, "data", x(:, i));
+      if (islogical (y))
+        y = double (y);
+      endif
       addproperty ("ydata", hg, "data", y(:, i));
       if (have_z)
         addproperty ("zdata", hg, "data", z(:, i));
       else
         addproperty ("zdata", hg, "data", []);
       endif
 
       addlistener (hg, "xdata", @update_data);
